integration/spark/src/main/scala/org/apache/spark/sql/execution/strategy/DDLStrategy.scala

```diff
@@ -204,6 +204,10 @@ class DDLStrategy(sparkSession: SparkSession) extends SparkStrategy {
           ExecutedCommandExec(CarbonCreateSecondaryIndexCommand(
             indexModel, tableProperties, ifNotExists, isDeferredRefresh, isCreateSIndex)) :: Nil
         } else {
+          if (!sparkSession.sessionState.catalog.
```

Comment:
already we are calling `tableExists `funtion to check whether carbon table or not, if the table not found, it throws `NoSuchTableException`. But we catch the exception in `org.apache.spark.sql.hive.CarbonFileMetastore#tableExists` and return false. So you can add an error log in CarbonFileMetaStore, saying the table not exists mentioning table name and avoid the calling to hive metastore here in your changes which is a costly operation.

Also, you can just change the error to more generalized way to satisfy the non-carbon table scenario and table not exists.

Reply:
done